### PR TITLE
Develop → Main 

### PR DESCRIPTION
**Pull request by ksookdhew**
Merging Dev into Main

### CODE REVIEW

**Review by AShozi — APPROVED** [review]

File: F1/F1/Constructor/ConstructorInfoTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+//
+
+import UIKit
+
+class ConstructorInfoTableViewCell: UITableViewCell {
+
+    // MARK: IBOutlets
+    @IBOutlet weak private var constructorImage: UIImageView!
+    @IBOutlet weak private var constructorName: UILabel!
+    @IBOutlet weak private var  constructorNationality: UILabel!
```
Comment (kieran-woodrow) [inline F1/F1/Constructor/ConstructorInfoTableViewCell.swift:15]: spaces seem off here

File: F1/F1/CoreData/CoreDataManager.swift [hunk 2]
```diff
@@ -0,0 +1,362 @@
+//
+//  CoreDataManager.swift
+//  F1
+//
+//  Created by Kaitlyn Sookdhew on 2024/05/15.
+//
+import CoreData
+import Foundation
```
Comment (kieran-woodrow) [inline F1/F1/CoreData/CoreDataManager.swift:8]: can remove foundation

File: F1/F1/CoreData/CoreDataManager.swift [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+import UIKit
+
+class CoreDataManager {
+
+    // MARK: Variables
+    static let shared = CoreDataManager()
```
Comment (kieran-woodrow) [inline F1/F1/CoreData/CoreDataManager.swift:14]: singleons should be avoided. You can just create n object of CoreDataManager(0 and use DI to inject it into the repo

File: F1/F1/Standings/StandigsViewModel.swift [hunk 4]
```diff
@@ -10,13 +10,21 @@ class StandingsViewModel {
 
     // MARK: Variables
     private weak var navigationDelegate: StandingsNavigationDelegate?
-    private let driverViewModel: DriverStandingsViewModel
-    private let constructorViewModel: ConstructorStandingsViewModel
+    private weak var delegate: ViewModelDelegate?
+    private var driverStanding: [DriverStanding]?
+    private var constructorStanding: [ConstructorStanding]?
+    let driverViewModel: DriverStandingsViewModel
```
Comment (kieran-woodrow) [inline F1/F1/Standings/StandigsViewModel.swift:16]: public then private

File: F1/F1/Standings/DriverStandings/DriverStandingsViewModel.swift [hunk 5]
```diff
@@ -10,14 +10,13 @@ import Foundation
 class DriverStandingsViewModel {
 
     // MARK: Variables
-    private weak var delegate: ViewModelDelegate?
     private var repository: DriverStandingsRepositoryType?
-    private var driverStanding: [DriverStanding]?
+    var driverStanding: [DriverStanding]?
```
Comment (kieran-woodrow) [inline F1/F1/Standings/DriverStandings/DriverStandingsViewModel.swift:14]: public then private

File: F1/F1/Config/AppDelegate.swift [hunk 6]
```diff
@@ -21,5 +40,15 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
     }
-}
 
+    // MARK: - Core Data stack
+    lazy var persistentContainer: NSPersistentContainer = {
+        let container = NSPersistentContainer(name: "F1Model")
+        container.loadPersistentStores(completionHandler: { (storeDescription, error) in
+            if let error = error as NSError? {
```
Comment (rvanderwesthuizen) [inline F1/F1/Config/AppDelegate.swift:48]:
```suggestion
            if let error as NSError? {
```

File: F1/F1/Standings/ConstructorStandings/ConstructorStandingsViewModel.swift [hunk 7]
```diff
@@ -9,13 +9,12 @@ import Foundation
 class ConstructorStandingsViewModel {
 
     // MARK: Variables
-    private weak var delegate: ViewModelDelegate?
     private var repository: ConstructorStandingsRepositoryType?
-    private var constructorStanding: [ConstructorStanding]?
+    var constructorStanding: [ConstructorStanding]?
```
Comment (kieran-woodrow) [inline F1/F1/Standings/ConstructorStandings/ConstructorStandingsViewModel.swift:13]: public then private

**Review by kieran-woodrow — APPROVED** [review]
please address comments in another pr

**Review by rvanderwesthuizen — COMMENTED** [review]
just the one comment and please have a look at the warnings

**Comment by ksookdhew** [comment]
Force merge by Lloyd

**Pull request merged**